File: stacked/stack.py
class Stack:
    def __init__(self) :
        self.stack=[]

    def push(self,x):
        self.stack.append(x)
    def pop(self):
        if self.is_empty():
            return False
        self.stack.pop()
    def top(self):
        if self.is_empty():
            return "Empty"
        return self.stack[-1]
    def is_empty(self):
        return len(self.stack)==0
st = Stack()
st.push(1)
st.push(2)
st.push(3)
st.push(4)
st.pop()
print(st.top())

File: stacked/stack.cpp
#include<bits/stdc++.h>
#include<iostream>
#include<algorithm>
#define n 100
using namespace std;

class Stack{
    int *arr;
    int top;

    public:

    stack(){
            arr= new int[n];
            top=-1;
    }

    void push(int x){
        if(top==n-1) {
            cout<<"Stack overflow\n";return;
        }
        top++;
        arr[top]=x;
    }

    void pop(){
        if(top==-1) {
            cout<<"Empty stack\n";
            return ;
        }

        top--;
    }

    void Top(){
         if(top==-1) {
            cout<<"Empty stack\n";
            return -1;
        } 

        return arr[top];
    }
    bool empty(){
        
        return top==-1;
    }

};78


int main()
{
    Stack st;
    st.push(1);
    st.push(2);
    st.push(3);
    st.push(4);
    st.pop();
    cout<<st.Top()<<endl;
 

return 0;
}